scripts/_utils.py

```
@@ -0,0 +1,171 @@
+
+    # Fallback: retry with shutil.rmtree
+    for attempt in range(3):
+        try:
+            shutil.rmtree(path, ignore_errors=False)
+
+            def handle_remove_readonly(func, path, _exc_info):
+                try:
+                    current_mode = os.stat(path).st_mode
+                    os.chmod(
+                        path,
+                        current_mode | stat.S_IWUSR | stat.S_IWGRP | stat.S_IWOTH,
```

Ruff (0.15.14)

[error] Undefined name `stat`

(F821)